Sources/Private/Experimental/Animations/TransformAnimations.swift

```diff
@@ -151,17 +151,41 @@ extension CALayer {
         // Lottie animation files express scale as a numerical percentage value
         // (e.g. 50%, 100%, 200%) so we divide by 100 to get the decimal values
         // expected by Core Animation (e.g. 0.5, 1.0, 2.0).
-        CGFloat(scale.x) / 100
+        //  - For some reason, negative `scaleX` values aren't applied by
+        //    Core Animation. Instead, we set up a `rotationY` animation below
+        //    to flip the view horizontally, which gives us the desired effect.
+        abs(CGFloat(scale.x) / 100)
```

Comment:
Do you think it makes sense to file a radar reporting this? Perhaps it could get fixed in the future by the CA team

Reply:
Good idea, this seems easy to create a repro case for

Reply:
Experimenting in a sample project, it looks like this is a consequence of how we set up a different `CAAnimation` for each component of the transform, instead of having a single `CAAnimation` for the entire transform.

For example, this animation correctly flips the layer on both vertically and horizontally:

```swift
let transform = CABasicAnimation(keyPath: "transform")
transform.duration = 1.0
transform.fromValue = CATransform3DMakeScale(-0.5, -0.5, 1)
transform.toValue = CATransform3DMakeScale(-1, -1, 1)
transform.repeatCount = .greatestFiniteMagnitude
transform.autoreverses = true
square.add(transform, forKey: "transform")
```

<img width="170" alt="Screen Shot 2022-01-28 at 2 33 17 PM" src="https://user-images.githubusercontent.com/1811727/151630665-db07f3d4-62c0-4ae8-b34e-4a7b3b105695.png">

but this seeming equivalent code causes the layer to only be flipped on one axis:

```swift
let scaleX = CABasicAnimation(keyPath: "transform.scale.x")
scaleX.duration = 1.0
scaleX.fromValue = -0.5
scaleX.toValue = -1
scaleX.repeatCount = .greatestFiniteMagnitude
scaleX.autoreverses = true
square.add(scaleX, forKey: "transform.scale.x")

let scaleY = CABasicAnimation(keyPath: "transform.scale.y")
scaleY.duration = 1.0
scaleY.fromValue = -0.5
scaleY.toValue = -1
scaleY.repeatCount = .greatestFiniteMagnitude
scaleY.autoreverses = true
square.add(scaleY, forKey: "transform.scale.y")
```

<img width="170" alt="Screen Shot 2022-01-28 at 2 31 15 PM" src="https://user-images.githubusercontent.com/1811727/151630408-756f548c-0d1f-47be-98c9-2caf1e84d7a7.png">

Reply:
Got an interesting response on the radar:

> Thanks for submitting this report.  This behaves as expected.
>
> There is inherent ambiguity in a transform matrix. For example, Scale(-0.5, -1, -1) * Rotation(PI, 1, 0, 0) is the same transform matrix as Scale(-0.5, 1, 1). 
>
> If scale.y is then set to -0.5, you would not get a y-flip in the first decomposition, and that’s the “issue” you observed. There’s no “fix” for this because whichever way we decide to decompose a transform matrix for merging multiple animations, one can always find a counterexample due to the ambiguity.

Reply:
Interesting! So I guess there's no way to specify this in a way that Core Animation will render—bummer.